monkey/monkey_island/cc/resources/propagation_credentials.py

```diff
@@ -35,7 +35,18 @@ def get(self, collection=None):
     @auth_token_required
     @roles_accepted(AccountRole.ISLAND_INTERFACE.name)
     def put(self, collection=None):
-        credentials = [Credentials(**c) for c in request.json]
+        try:
+            credentials = []
+            for index, credential_pair in enumerate(request.json):
+                credentials.append(Credentials(**credential_pair))
+        except (TypeError, ValueError) as err:
+            return {
+                "error": {
+                    "message": f"{str(err)}",
+                    "bad_index": index,
```

Comment:
"bad" is redundant. This is an error message.
```suggestion
                    "index": index,
```